lib/modules/platform/gerrit/scm.ts

```diff
@@ -0,0 +1,161 @@
+import { randomUUID } from 'crypto';
+import { logger } from '../../../logger';
+import type { PrState } from '../../../types';
+import * as git from '../../../util/git';
+import type { CommitFilesConfig, CommitSha } from '../../../util/git/types';
+import { toSha256 } from '../../../util/hasha';
+import { DefaultGitScm } from '../default-scm';
+import { client } from './client';
+import { mapPrStateToGerritFilter } from './utils';
+
+let repository: string;
+let username: string;
+export const configureScm = (repo: string, login: string): void => {
+  repository = repo;
+  username = login;
+};
+
+export class GerritScm extends DefaultGitScm {
+  private createFilter(
+    state: PrState,
+    branchName: string,
+    baseBranch?: string
+  ): string[] {
+    const filterState = mapPrStateToGerritFilter(state);
+    const filter = ['owner:self', 'project:' + repository, filterState];
+    filter.push(`hashtag:sourceBranch-${branchName}`);
+    if (baseBranch) {
+      filter.push(`branch:${baseBranch}`);
+    }
+    return filter;
+  }
+
+  override async branchExists(branchName: string): Promise<boolean> {
+    const filter = this.createFilter('open', branchName);
+    const change = await client.findChanges(filter).then((res) => res.pop());
+    if (change) {
+      return true;
+    }
+    return git.branchExists(branchName);
+  }
+
+  override async getBranchCommit(
+    branchName: string
+  ): Promise<CommitSha | null> {
+    const filter = this.createFilter('open', branchName);
+    const change = await client.findChanges(filter).then((res) => res.pop());
+    if (change) {
+      return change.current_revision!;
+    }
+    return git.getBranchCommit(branchName);
+  }
+
+  override async isBranchBehindBase(
+    branchName: string,
+    baseBranch: string
+  ): Promise<boolean> {
+    const filter = this.createFilter('open', branchName, baseBranch);
+    const change = await client
+      .findChanges(filter, true)
+      .then((res) => res.pop());
+    if (change) {
+      const currentGerritPatchset = change.revisions![change.current_revision!];
+      return currentGerritPatchset.actions?.['rebase'].enabled === true;
+    }
+    return true;
+  }
+
+  override async isBranchConflicted(
+    baseBranch: string,
+    branch: string
+  ): Promise<boolean> {
+    const filter = this.createFilter('all', branch, baseBranch);
+    const change = await client.findChanges(filter).then((res) => res.pop());
+    if (change) {
+      const mergeInfo = await client.getMergeableInfo(change);
+      return !mergeInfo.mergeable;
+    } else {
+      //TODO: is this correct? what about closed changes? (HTTP-Status=409 from getMergeableInfo)
```

Comment:
I'm not completely sure what you mean with "shouldn't be called...", but i think you misinterpret my (poorly worded) TODO.

The `findChanges` call with `state=all` will search for all types of (open, merged or abandoned) changes.
But the call to `getMergeableInfo` will fail with http-code 409 if the change is not open anymore (already merged or abandoned).
I will change this back to `findChanges('open', ...)` and return true if no change could be found (because merged, abandoned or never existed) as you suggested.